### ISSUE TITLE
Keep output signature of the test fn on #[wasm_bindgen_test]

### ISSUE THREAD

**Issue by gurobokum** [rustwasm/wasm-bindgen#1830]
### Motivation

I would like to [use ?](https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/question-mark-in-main-and-tests.html) in my wasm_bindgen tests

### Proposed Solution
```
#[wasm_bindgen_test]
fn capture_keypress() -> Result<(), JsValue> {
   ....
   let event = document.create_event("KeyboardEvent")?;
   ...
    Ok(())
}
```
should correctly work, now I"m getting
```
error[E0271]: type mismatch resolving `<fn() -> std::result::Result<(), wasm_bindgen::JsValue> {capture_keypress} as std::ops::FnOnce<()>>::Output == ()`
 --> lib/system/web/tests/keyboard_engine.rs:9:1
  |
9 | #[wasm_bindgen_test]
  | ^^^^^^^^^^^^^^^^^^^^ expected enum `std::result::Result`, found ()
  |
  = note: expected type `std::result::Result<(), wasm_bindgen::JsValue>`
             found type `()`
```

**Comment by alexcrichton** [comment]
Thanks for the report! Support this seems pretty reasonable to me!

Right now the `#[wasm_bindgen_test]` macro is [pretty simple](https://github.com/rustwasm/wasm-bindgen/blob/e880baf11ddd49f7d961de16a7e28a85199c58e6/crates/test-macro/src/lib.rs), but we can probably support this with some trait trickery.

**Comment by daxpedda** [comment]
Resolved by #3051.
